src/lib/locale/locales.js

```diff
@@ -143,15 +143,21 @@ export function defineLocale (name, config) {
 export function updateLocale(name, config) {
     if (config != null) {
         var locale, tmpLocale, parentConfig = baseConfig;
-        // MERGE
-        tmpLocale = loadLocale(name);
-        if (tmpLocale != null) {
-            parentConfig = tmpLocale._config;
+
+        if (locales[name] != null && locales[name].parentLocale != null) {
+            // Update existing child locale in-place to avoid memory-leaks
+            locales[name].set(config);
```

Comment:
Does this require the config provided to `updateLocale` to be a complete config?

Reply:
See, for example, https://github.com/moment/moment/issues/4438

Reply:
Good question... just because it doesn't break current unit-tests doesn't mean it's any good :-/

It should probably needs to be slightly more hideous, like:
```js
locales[name].set(mergeConfigs(locales[name]._config, config));
```

I'll try that when I get a chance.